### PR TITLE
Fixes abductor vest showing when something is in suit storage

### LINKED ISSUE #38058: Putting an abductor baton on your suit slot while disguised makes the ayy vest visible
It changes the sprite from your disguise to the vest, thus ruining the disguise until you re-switch out of the disguise and back in

### PR DESCRIPTION
**Pull request by CitrusGender**
Just because something is inside of a vest doesn't mean it should show the vest. This only applies to outerwear since that's the most outwardly apparent. I suppose it could be added to other things but that's just debatable. 
:cl: 
fix: The abductor vest (and now all suits/vests) overlay will no longer be shown when something is in storage.
/:cl:

[why]: # Fixes #38058

### CODE REVIEW

**Comment by CitrusGender** [comment]
is this really a feature

**Comment by YakumoChen** [comment]
it automatically adds the tag when you use the add: prefix in the CL

**Comment by ExcessiveUseOfCobblestone** [comment]
Is there no better way to incorporate this? Adding a var on every human mob and checking for it every time you have something in the suit slot when you update icons for a single niche instance doesn't sit right. Then again this is probably legacy code so only RNGesus knows.

**Comment by MrDoomBringer** [comment]
Citrus you probably dont have to worry about the "Adds an is_disguised variable that turns on and off when disguised with the abductor vest." line in your changelog, as _generally_ we only show player-facing features, unless its unique code that we want coders to make use of or something.

**Comment by CitrusGender** [comment]
@ExcessiveUseOfCobblestone It's better than checking to see if that specific item is on (hence why I added the variable.) It is a problem for all suits if you cut overlays so this essentially provides a check that can be used for other suits that cut overlays.

**Comment by CitrusGender** [comment]
also, as a note, I can just remove the code that causes the suit layer overlay to be changed when the appearance is updated and we won't need a variable then @KorPhaeron . Again, it's a question if whether or not that was intentional.

**Review by optimumtact — DISMISSED** [review]
as mentioned in discord/irc the correct way to fix this is to fix the implementation relying on overlays not being rebuilt on the suit

**Comment by CitrusGender** [comment]
@optimumtact so I took a second look at this again and changing the invisibility (as you recommended) will not show the new overlay via M.add_overlay(disguise.overlays) if I'm not mistaken. Is there a way to create overlays over an invisible mob then?

**Comment by optimumtact** [comment]
make the mob invisible and then add a temp effect that follows the mob until the disguise is over. Clicks on the effect should fall through to the mob, so it's just to hold the overlays.

however if the mob is visible and doesnt' need to be invisible (i.e it's just changing how it looks) then you should be using alternate appearance code anyway

**Comment by CitrusGender** [comment]
https://feen.us/sgeghovq.mp4
so uh, there's really no correct way to do this since the alternative_appearance code doesn't have directions. I tried in the video linked above. Maybe another maintainer has no qualms with this edit, however.
image

**Comment by AnturK** [comment]
This isn't a fix this is removing unrelated code.

File: code/modules/mob/living/carbon/human/update_icons.dm
Comment (AnturK) [inline code/modules/mob/living/carbon/human/update_icons.dm]: Another issue is right here, it was applying whole suit sprite instead of just belt sprite.

Comment (CitrusGender) [reply code/modules/mob/living/carbon/human/update_icons.dm]: Yeah, that's why I removed it. Also, wasn't the subsequent code relying upon the variable that was just defined? This would make line 305 through 308 useless because all it does is adjust the suit layer. Additionally, it wouldn't affect the belt layer because it's inside the suit.

**Comment by ExcessiveUseOfCobblestone** [comment]
@Robustin not sure how invested you're still in regarding overlays but perhaps you can put that noggin to use here :^)

**Comment by CitrusGender** [comment]
I'm waiting for @AnturK 's reply.

File: code/modules/mob/living/carbon/human/update_icons.dm
Comment (AnturK) [reply code/modules/mob/living/carbon/human/update_icons.dm]: if you want to remove the feature, open proper removal PR because this only cuts a part of it without cleaning it up, if you want to fix the typo this this is not a it.

**Comment by AnturK** [comment]
Feel free to open proper fix for this or removal of suit storage layer.

**Pull request closed without merging**
